### PR TITLE
Publish extension via GH action on GH release

### PR DESCRIPTION
**Pull request by juliasilge**
This PR sets up infrastructure to publish the extension to the two marketplaces (OpenVSX and the Visual Studio Marketplace) via GH action via doing a GH _release_. I'll make a few notes on the PR for us to use for discussion moving forward.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -2,6 +2,9 @@
 name: Build VS Code Extension
 
 on:
+  release:
+    types:
+      - published
```
Comment (juliasilge) [inline .github/workflows/ci.yaml:7]: I set this up to use a _release_ but alternatively we could [use a tag in the way the Shiny team does](https://github.com/posit-dev/shiny-vscode/blob/main/.github/workflows/publish.yml).

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -37,3 +40,34 @@ jobs:
...
+          node-version: latest
+      - run: yarn install --immutable --immutable-cache --check-cache
+      - uses: actions/download-artifact@v4
+        with:
+          name: quarto-vscode-${{ github.sha }}
+
+      - name: Publish to Open VSX Registry
+        uses: HaaLeo/publish-vscode-extension@v1
+        with:
+          pat: ${{ secrets.OPEN_VSX_TOKEN }}
```
Comment (juliasilge) [inline .github/workflows/ci.yaml:65]: This is the first of two tokens we will need added to the repo secrets to make this work.

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -37,3 +40,34 @@ jobs:
...
+          extensionFile: ${{ needs.package_extension.outputs.vsixPath }}
+
+      - name: Publish to Visual Studio Marketplace
+        uses: HaaLeo/publish-vscode-extension@v1
+        with:
+          pat: ${{ secrets.VS_MARKETPLACE_TOKEN }}
```
Comment (juliasilge) [inline .github/workflows/ci.yaml:71]: This is the second of two tokens we will need added to the repo secrets to make this work.

**Review by juliasilge — COMMENTED** [review]
This action builds the extension one time and then publishes exactly the same file to both OpenVSX and the Microsoft marketplace. There are alternative approaches if we have a reason for another approach.

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (juliasilge) [reply .github/workflows/ci.yaml:65]: JJ added this a couple of days ago to this repo's secrets. ✅

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (juliasilge) [reply .github/workflows/ci.yaml:71]: I just added this to this repo's secrets, after being added as a publisher for the Quarto extension. I made a token that expires after 90 days FYI so we will need to replace it at that time. If we feel comfortable, next time I can make one that never expires.

**Comment by cscheid** [comment]
This all looks good to me!

**Pull request merged**